api/METRICS.md

```diff
@@ -35,72 +45,260 @@ acc_ratio = accuracy_score_ratio(y_true, y_pred, sensitive_features=sf, **other_
 acc_group_min = accuracy_score_group_min(y_true, y_pred, sensitive_features=sf, **other_kwargs)
 ```
 
-## Proposal
+## Proposal details
 
-*Function signatures*
+The items that are not implemented yet are marked as `[TODO]`.
+
+### Metrics engine
 
 ```python
 group_summary(metric, y_true, y_pred, *, sensitive_features, **other_kwargs)
 # return the group summary for the provided `metric`, where `metric` has the signature
 # metric(y_true, y_pred, **other_kwargs)
 
-make_metric_group_summary(metric)
+make_group_summary(metric) [TODO]
 # return a callable object <metric>_group_summary:
-# <metric>_group_summary(...) = group_summary(<metric>, ...)
+# <metric>_group_summary(...) = group_summary(metric, ...)
 
-# Transformation functions returning scalars
+# Transformation functions returning scalars (the definitions are below)
 difference_from_summary(summary)
 ratio_from_summary(summary)
 group_min_from_summary(summary)
 group_max_from_summary(summary)
 
-# Metric-specific functions returning group summary and scalars
+derived_metric(metric, transformation, y_true, y_pred, *, sensitive_features, **other_kwargs) [TODO]

```

Comment:
What's the use case for this? Doesn't it just wrap `difference_from_summary(make_group_summary(....))` ?

Reply:
Or is this evaluating things "all the way", whereas the next `make_derived_metric` returns the callable object?

Reply:
Yes... this is to evaluate the scalar whereas `make_derived_metric` is just returning the wrapper. the relationship between `derived_metric` and `make_derived_metric` is exactly the same as between `group_summary` and `make_group_summary`. Let me tweak the explanation.